osquery/tables/system/darwin/homebrew_packages.cpp

```diff
@@ -62,6 +74,162 @@ std::vector<std::string> getHomebrewVersionsFromInfoPlistPath(
+  while (std::getline(file, line)) {
+    if (std::regex_match(line, auto_updates_regex)) {
+      return true;
+    }
+  }
+
+  return false;
+}
+
+// Looking for the value of "auto_updates" in a .json file
+// Defaults to false if the key is not found
+bool getBooleanValueFromJsonFile(const std::string& file_path,
+                                 const std::string& key = "auto_updates") {
+  std::string content;
+  auto status = osquery::readFile(file_path, content);
+  if (!status.ok()) {
+    TLOG << "Error reading file " << file_path << ": " << status.toString();
+    return false;
+  }
+
+  pt::ptree tree;
+  std::istringstream is(content);
+  try {
+    pt::read_json(is, tree);
+  } catch (const pt::json_parser_error& e) {
+    TLOG << "Error parsing JSON from file " << file_path << ": " << e.what();
+    return false;
+  }
+
+  return tree.get<bool>(key, false);
+}
+
+bool getHomebrewAutoUpdate(const std::string& path) {
+  // a Cask full path looks typically like
+  // /opt/homebrew/Caskroom/iterm2/3.5.9/iTerm.app -> /Applications/iTerm.app
+  //
+  // The path passed to this function would be /opt/homebrew/Caskroom/iterm2
+  //
+  // Auto update data generally exists in the .metadata directory
+  // Under a json OR an .rb file with the cask name:
+  // /opt/homebrew/Caskroom//iterm2/.metadata/3.5.9/20241116155943.669/Casks/iterm2.json
+  // /opt/homebrew/Caskroom//vlc/.metadata/3.0.18/20230607170348.510/Casks/vlc.rb
+
+  // https://github.com/Homebrew/brew/blob/41be66fc2a3facc5a575f98ffa2a17323ef30c22/Library/Homebrew/cask/cask.rb#L175
+  // (cask.rb#L175)
+  // https://github.com/Homebrew/brew/blob/41be66fc2a3facc5a575f98ffa2a17323ef30c22/Library/Homebrew/cask/installer.rb#L402-L403
+  // (installer.rb#L402-L403)
+
+  std::vector<std::string> files;
+
+  std::string app_name = fs::path(path).filename().string();
+
+  auto metadata_path = path + "/.metadata";
+  // Looking for the config file in the .metadata directory by
+  // listing all files in the directory
+  auto status =
+      osquery::listFilesInDirectory(metadata_path, files, true /* recursive */);
+  if (!status.ok()) {
+    TLOG << "Error listing files in " << metadata_path << ": "
+         << status.toString();
+    return false;
+  }
+  // If the config file contains "auto_updates true", we can return early
+  // as we know the cask auto updates
+  for (const auto& file : files) {
+    // iterate over all files in the .metadata directory
+    // and only consider .json and .rb files with the app name
+    std::string filename = fs::path(file).filename().string();
+    if (filename == app_name + ".json") {
+      return getBooleanValueFromJsonFile(file);
+    }
+    if (filename == app_name + ".rb") {
+      return checkAutoUpdatesInFile(file);
+    }
+  }
+  // If we didn't find the config file in the .metadata directory, we default to
+  // false (should we return an error instead?)
+  TLOG << "No config file found in  " << metadata_path;
+  return false;
```

Comment:
@directionless : I'm wondering what behavior would be best if no metadata file has been found. Any idea ?

Reply:
What does that mean for the underlying state?

If it's normal, don't populate the column. If it's highly unusual, consider logging. But if it's so common that every machine has a couple, that's a lot of log noise...

Reply:
I'm actually not sure. I mostly did a bunch of reverse engineering on local homebrew installs, and am pretty confident that this `.metadata` directory should always be there, so logging in case of not finding it seems like the sensible thing to do 🤔